cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1344 @@
+func Add(fileName, exchName, checkType, path string, data interface{}, update bool) error {
+	finalResp, check, err := CheckExistingExchanges(jsonFile, exchName)
+	if err != nil {
+		return err
+	}
+	var file []byte
+	switch update {
+	case false:
+		if check {
+			if verbose {
+				log.Printf("%v exchange already exists\n", exchName)
+			}
+			return nil
+		}
+		exchange, err := FillData(exchName, checkType, path, data)
+		if err != nil {
+			return err
+		}
+		finalResp = append(finalResp, exchange)
+		file, err = json.MarshalIndent(finalResp, "", " ")
+		if err != nil {
+			return err
+		}
+	case true:
+		info, err := FillData(exchName, checkType, path, data)
+		if err != nil {
+			return err
+		}
+		allExchData, err := Update(exchName, finalResp, info)
+		if err != nil {
+			return err
+		}
+		file, err = json.MarshalIndent(allExchData, "", " ")
+		if err != nil {
+			return err
+		}
+	}
+	return ioutil.WriteFile(jsonFile, file, 0770)
+}
+
+// FillData fills exchange data based on the given checkType
+func FillData(exchName, checkType, path string, data interface{}) (ExchangeInfo, error) {
+	switch checkType {
+	case github:
+		tempSha, err := getSha(path)
+		if err != nil {
+			return ExchangeInfo{}, err
+		}
+		return ExchangeInfo{
+			Name:      exchName,
+			CheckType: checkType,
+			Data: &CheckData{
+				GitHubData: &GithubData{
+					Repo: path,
+					Sha:  tempSha.ShaResp},
+			},
+		}, nil
+	case htmlScrape:
+		tempData := data.(HTMLScrapingData)
+		checkStr, err := CheckChangeLog(tempData)
+		if err != nil {
+			return ExchangeInfo{}, err
+		}
+		return ExchangeInfo{
+			Name:      exchName,
+			CheckType: checkType,
+			Data: &CheckData{
+				HTMLData: &HTMLScrapingData{
+					CheckString:   checkStr,
+					DateFormat:    tempData.DateFormat,
+					Key:           tempData.Key,
+					RegExp:        tempData.RegExp,
+					TextTokenData: tempData.TextTokenData,
+					TokenData:     tempData.TokenData,
+					TokenDataEnd:  tempData.TokenDataEnd,
+					Val:           tempData.Val,
+					Path:          tempData.Path},
+			},
+		}, nil
+	default:
+		return ExchangeInfo{}, errors.New("invalid checkType")
+	}
+}
+
+// HTMLScrapeDefault gets check string data for the default cases
+func HTMLScrapeDefault(htmlData HTMLScrapingData) ([]string, error) {
+	var resp []string
+	temp, err := http.Get(htmlData.Path)
+	if err != nil {
+		return resp, err
+	}
+	tokenizer := html.NewTokenizer(temp.Body)
+loop:
+	for {
+		next := tokenizer.Next()
+		switch next {
+		case html.ErrorToken:
+			break loop
+		case html.StartTagToken:
+			token := tokenizer.Token()
+			if token.Data == htmlData.TokenData {
+				for _, a := range token.Attr {
+					if a.Key == htmlData.Key && a.Val == htmlData.Val {
+					loop2:
+						for {
+							nextToken := tokenizer.Next()
+							switch nextToken {
+							case html.EndTagToken:
+								t := tokenizer.Token()
+								if t.Data == htmlData.TokenDataEnd {
+									break loop2
```

Comment:
Same with here, you can just use `continue`
These kind of breaks are useful when getting out of higher level loops (like say you wanted to iterate over `loop`), however, since you're just wanting to continue the current, innermost loop, `continue` still works fine